lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -452,6 +452,19 @@ class NetworkAnalyzer {
     // The main document is the earliest document request, using position in networkRecords array to break ties.
     return documentRequests.reduce((min, r) => (r.startTime < min.startTime ? r : min));
   }
+
+  /**
+   * Resolves redirect chain given a main document.
+   * See: {@link NetworkAnalyzer.findMainDocument}) for how to retrieve main document.
+   *
+   * @param {LH.Artifacts.NetworkRequest|undefined} request
+   * @returns {LH.Artifacts.NetworkRequest|undefined}
+   */
+  static resolveRedirects(request) {
```

Comment:
nit: @patrickhulce would this method be better on NetworkRequest?

Reply:
Marginally yes, but we have network analysis somewhat scattered at the moment, so I don't think it makes a large difference.